libcxx/src/support/runtime/exception_pointer_cxxabi.ipp

```diff
@@ -36,6 +29,21 @@ exception_ptr exception_ptr::__from_native_exception_pointer(void* __e) noexcept
   return ptr;
 }
 
+exception_ptr::~exception_ptr() noexcept { __decrement_refcount(__ptr_); }
+
+exception_ptr::exception_ptr(const exception_ptr& other) noexcept : __ptr_(other.__ptr_) {
+  __increment_refcount(__ptr_);
+}
+
+exception_ptr& exception_ptr::operator=(const exception_ptr& other) noexcept {
+  if (__ptr_ != other.__ptr_) {
+    __increment_refcount(other.__ptr_);
+    __decrement_refcount(__ptr_);
+    __ptr_ = other.__ptr_;
+  }
+  return *this;
```

Comment:
Can we maybe merge these implementations now?

Reply:
merged into `exception_pointer_refcounted.ipp`. Not 100% happy with the name, would be open to a more speaking name

I realized that `__from_native_exception_pointer` can also be merged across 3 ABIs and could also be inlined into the header. I think this would only give us marginal performance savings, though. Should we still do it for other reasons, e.g. consistency?